### DISCUSSION TITLE
Merging Pushing Pulling All That 😭

### DISCUSSION THREAD

**Discussion by alizezila** [albertkun/21S-ASIAAM-191A#156] [Help Desk] [answered]
Hello everyone! I was wondering if someone could help me out, I am thoroughly confused. My group-mate created a repo for our group project and I cloned the repo and created a new branch. So now I am working on the new branch (I did git checkout "nameofnewbranch" so I know im working on that one) and when I make changes I commit them and push them, but I dont see the changes on the main branch. How do i fix this? I don't fully understand pulling and merging and feel like I am doing it all wrong. Can provide more detail if needed

**Comment by albertkun** [comment, marked as answer]
Hello @alizezila once you finish your changes, you have to make a pull request to merge them back into the main branch! i'll try to explain this a little bit more tomorrow in class, but here's a helpful guide:

https://medium.com/@urna.hybesis/pull-request-workflow-with-git-6-steps-guide-3858e30b5fa4

I think here is the picture of where you are at:

![image](https://user-images.githubusercontent.com/8574425/119749182-6a4a2200-be4b-11eb-87ff-205f9b19fa58.png)

Please reach out to me if you are still stuck though!

**Reply by alizezila** [reply]
this helped thank you! you would think after 10 weeks I would know by now..

**Reply by albertkun** [reply]
@alizezila  noooo, i didnt cover merges fully, so its on mel!!!
